### ISSUE TITLE
[Ruby] Implement new Concept Exercise: nil

### ISSUE THREAD

**Issue by biancapower** [exercism/v3#1918]
This issue describes how to implement the `nil` concept exercise for the Ruby track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of nil in Ruby.

## Learning objectives

- Know that `nil` is a special Ruby object used to represent an "empty" or "default" value
- Know that `nil` is falsey

## Out of scope

- `NilClass`

## Concepts

- `nil`
- `nil?` method

## Prerequisites

- `basics`
- `booleans`

### Hints

- [Data Types in Ruby - True, False, and Nil Explained with Examples](https://www.freecodecamp.org/news/data-types-in-ruby-true-false-and-nil-explained-with-examples/)

### After

- [The nil value in Ruby](https://medium.com/rubycademy/the-nil-value-in-ruby-d60e6a3642b9#:~:text=method%20implementation-,The%20nil%20value,%E2%80%9Clack%20of%20an%20object%E2%80%9D.&text=Unlike%20other%20languages%2C%20the%20nil,the%20non%2Dinstantiable%20NilClass%20class.)

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/ruby-representer).

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/ruby-analyzer).

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/ruby/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by neenjaw** [comment]
@biancapower do you think I could help with this exercise?

I would like to port over [elixir's `nil` exercise](https://github.com/exercism/v3/tree/master/languages/elixir/exercises/concept/nil)

**Comment by biancapower** [comment]
@neenjaw please do!

**Comment by neenjaw** [comment]
See #2046, as it is the follow-up issue.
